demos/free_surface/test_viscous_surface.py

```diff
@@ -0,0 +1,94 @@
+import numpy as np
+from pathlib import Path
+import pytest
+
+base = Path(__file__).parent.resolve()
+
+
+def run_benchmark(model):
+    # Run default case run for four dt factors
+    if model.name == 'explicit':
+        dtf_start = 1
+        iterative = False
+    else:
+        dtf_start = 2
+        if model.name == 'implicit-cylindrical-iterative':
+            iterative = False  # Cylinder cases already default to iterative so don't need to test this again!
+        else:
+            iterative = True
+
+    dt_factors = dtf_start / (2**np.arange(4))
+
+    errors = []
+    errors_zeta = []
+
+    for dtf in dt_factors:
+        simulation = model(dtf)
+        simulation.run_simulation()
+        errors.append(simulation.final_error)
+
+        if simulation.bottom_free_surface:
+            errors_zeta.append(simulation.final_zeta_error)
+
+    if simulation.bottom_free_surface:
+        np.savetxt(f"errors-{model.name}-top-free-surface-coupling.dat", errors)
+        np.savetxt(f"errors-{model.name}-bottom-free-surface-coupling.dat", errors_zeta)
+    else:
+        np.savetxt(f"errors-{model.name}-free-surface-coupling.dat", errors)
+
+    if iterative:
+        # Rerun with iterative solvers
+        errors_iterative = []
+        errors_zeta_iterative = []
+
+        for dtf in dt_factors:
+            simulation = model(dtf, iterative_2d=True)
+            simulation.run_simulation()
+            errors_iterative.append(simulation.final_error)
+
+            if simulation.bottom_free_surface:
+                errors_zeta_iterative.append(simulation.final_zeta_error)
+
+        if simulation.bottom_free_surface:
+            np.savetxt(f"errors-{model.name}-iterative-top-free-surface-coupling.dat", errors_iterative)
+            np.savetxt(f"errors-{model.name}-iterative-bottom-free-surface-coupling.dat", errors_zeta_iterative)
+        else:
+            np.savetxt(f"errors-{model.name}-iterative-free-surface-coupling.dat", errors_iterative)
```

Comment:
cool thanks Angus, as discussed I've used this structure and added direct/iterative attributes to all the cases so that it skips the the explicit iterative case and the direct cylindrical case.